opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/aggregation/ExpressionAggregationScriptTest.java

```diff
@@ -103,6 +124,45 @@ void can_execute_parse_expression() {
         .shouldMatch("30");
   }
 
+  @Test
+  void can_execute_expression_interpret_dates_for_aggregation() {
+    assertThat()
+        .docValues("date", "1961-04-12")
+        .evaluate(
+            dsl.date(ref("date", STRING)))
+        .shouldMatch(new ExprDateValue(LocalDate.of(1961, 4, 12))
+            .timestampValue().toEpochMilli());
+  }
+
+  @Test
+  void can_execute_expression_interpret_datetimes_for_aggregation() {
+    assertThat()
+        .docValues("datetime", "1984-03-17 22:16:42")
+        .evaluate(
+            dsl.datetime(ref("datetime", STRING)))
+        .shouldMatch(new ExprDatetimeValue("1984-03-17 22:16:42")
+            .timestampValue().toEpochMilli());
+  }
+
+  @Test
+  void can_execute_expression_interpret_times_for_aggregation() {
+    assertThat()
+        .docValues("time", "22:13:42")
+        .evaluate(
+            dsl.time(ref("time", STRING)))
+        .shouldMatch(MILLIS.between(LocalTime.MIN, LocalTime.of(22, 13, 42)));
+  }
+
+  @Test
+  void can_execute_expression_interpret_timestamps_for_aggregation() {
+    assertThat()
+        .docValues("timestamp", "1984-03-17 22:16:42")
+        .evaluate(
+            dsl.timestamp(ref("timestamp", STRING)))
+        .shouldMatch(new ExprTimestampValue("1984-03-17 22:16:42")
+            .timestampValue().toEpochMilli());
+  }
+
```

Comment:
should we add the all-nulls case here too?

Reply:
I'm planning to add ITs that cover too.

Reply:
Existing UT framework is very limited and can't properly cover such case. There is a test for that and it didn't detect such case:
https://github.com/Bit-Quill/opensearch-project-sql/blob/3eba57e275dc479ee9a7771bf912aed80a754a25/opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/aggregation/ExpressionAggregationScriptTest.java#L102-L108